### ISSUE TITLE
JDK 9 and 1.2-SNAPSHOT producing error

### ISSUE THREAD

**Issue by sormuras** [google/google-java-format#86]
With 1.1 all works well.  This is the stack trace produced:

```
java.lang.IllegalAccessError: class com.google.googlejavaformat.java.Formatter (in unnamed module @0x6785e955) cannot access class com.sun.tools.javac.util.Context (in module jdk.compiler) because module jdk.compiler does not export com.sun.tools.javac.util to unnamed module @0x6785e955
at com.google.googlejavaformat.java.Formatter.format(Formatter.java:133)
at com.google.googlejavaformat.java.Formatter.getFormatReplacements(Formatter.java:237)
at com.google.googlejavaformat.java.Formatter.formatSource(Formatter.java:215)
at com.google.googlejavaformat.java.FormatFileCallable.call(FormatFileCallable.java:46)
at com.google.googlejavaformat.java.FormatFileCallable.call(FormatFileCallable.java:27)
at java.util.concurrent.FutureTask.run(java.base@9-ea/FutureTask.java:264)
at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@9-ea/ThreadPoolExecutor.java:1158)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@9-ea/ThreadPoolExecutor.java:632)
at java.lang.Thread.run(java.base@9-ea/Thread.java:843)
```

Who is using `com.sun.tools.javac.util.Context` [now](https://github.com/google/google-java-format/commit/ba2a958fc7b2452779f99c9246ee0feb07f52815#diff-c96edae3736870de40b2ab8b0657014dR38)?

**Comment by cushon** [comment]
Yep, there's some work to be done before the next release to make sure ba2a958fc7b2452779f99c9246ee0feb07f52815 plays nicely with everything.

**Comment by sormuras** [comment]
Mh, locally, I added the [Apache Maven JDeps Plugin](https://maven.apache.org/components/plugins/maven-jdeps-plugin) to the build -- and it does not fail. It just produces this trace:
```
[INFO] ------------------------------------------------------------------------
[INFO] Building Google Java Format 1.2-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-jdeps-plugin:3.0.0:jdkinternals (default-cli) @ google-java-format ---
[INFO] 
classes -> ~\.m2\repository\com\google\errorprone\error_prone_annotations\2.0.8\error_prone_annotations-2.0.8.jar
classes -> ~\.m2\repository\com\google\guava\guava\19.0\guava-19.0.jar
classes -> ~\.m2\repository\com\google\errorprone\javac\1.9.0-dev-r2973-2\javac-1.9.0-dev-r2973-2.jar
classes -> ~\.m2\repository\org\eclipse\equinox\org.eclipse.equinox.common\3.6.0\org.eclipse.equinox.common-3.6.0.jar
classes -> ~\.m2\repository\org\eclipse\jdt\org.eclipse.jdt.core\3.10.0\org.eclipse.jdt.core-3.10.0.jar
classes -> ~\jdk1.8.0_102\jre\lib\rt.jar
   com.google.googlejavaformat (classes)
      -> com.google.common.base                             guava-19.0.jar
      -> com.google.common.collect                          guava-19.0.jar
      -> java.io                                            
      -> java.lang                                          
      -> java.util                                          
      -> javax.tools                                        javac-1.9.0-dev-r2973-2.jar
   com.google.googlejavaformat.java (classes)
      -> com.google.common.base                             guava-19.0.jar
      -> com.google.common.collect                          guava-19.0.jar
      -> com.google.common.io                               guava-19.0.jar
      -> com.google.errorprone.annotations                  error_prone_annotations-2.0.8.jar
      -> com.google.googlejavaformat                        classes
      -> com.google.googlejavaformat.java.javadoc           classes
      -> com.sun.source.tree                                javac-1.9.0-dev-r2973-2.jar
      -> com.sun.source.util                                javac-1.9.0-dev-r2973-2.jar
      -> com.sun.tools.javac.code                           javac-1.9.0-dev-r2973-2.jar
      -> com.sun.tools.javac.file                           javac-1.9.0-dev-r2973-2.jar
      -> com.sun.tools.javac.parser                         javac-1.9.0-dev-r2973-2.jar
      -> com.sun.tools.javac.tree                           javac-1.9.0-dev-r2973-2.jar
      -> com.sun.tools.javac.util                           javac-1.9.0-dev-r2973-2.jar
      -> java.io                                            
      -> java.lang                                          
      -> java.net                                           
      -> java.nio.charset                                   
      -> java.nio.file                                      
      -> java.util                                          
      -> java.util.concurrent                               
      -> java.util.regex                                    
      -> javax.lang.model.element                           javac-1.9.0-dev-r2973-2.jar
      -> javax.lang.model.type                              javac-1.9.0-dev-r2973-2.jar
      -> javax.tools                                        javac-1.9.0-dev-r2973-2.jar
      -> org.eclipse.core.runtime                           org.eclipse.equinox.common-3.6.0.jar
      -> org.eclipse.jdt.core                               org.eclipse.jdt.core-3.10.0.jar
      -> org.eclipse.jdt.core.compiler                      org.eclipse.jdt.core-3.10.0.jar
      -> org.eclipse.jdt.core.dom                           org.eclipse.jdt.core-3.10.0.jar
   com.google.googlejavaformat.java.filer (classes)
      -> com.google.common.base                             guava-19.0.jar
      -> com.google.common.io                               guava-19.0.jar
      -> com.google.googlejavaformat.java                   classes
      -> java.io                                            
      -> java.lang                                          
      -> javax.annotation.processing                        javac-1.9.0-dev-r2973-2.jar
      -> javax.lang.model.element                           javac-1.9.0-dev-r2973-2.jar
      -> javax.tools                                        javac-1.9.0-dev-r2973-2.jar
   com.google.googlejavaformat.java.javadoc (classes)
      -> com.google.common.base                             guava-19.0.jar
      -> com.google.common.collect                          guava-19.0.jar
      -> com.google.googlejavaformat.java                   classes
      -> java.lang                                          
      -> java.util                                          
      -> java.util.regex
```
a) How do I convince the J9 runtime to work with GJF 1.2?
b) Shall I file a PR with the JDeps added to GJFs' `pom.xml`?

**Comment by sormuras** [comment]
Self-reply to a) via http://openjdk.java.net/jeps/261

    -XaddExports:jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED
    -XaddExports:jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED
    -XaddExports:jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED
    -XaddExports:jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED

**Comment by sormuras** [comment]
If I'm not mistaken, the only issue is the missing `module-info.java` configuration in GJF.

c) Change the issue title to **module-info.java missing**?

**Comment by cushon** [comment]
This was resolved by 4f29b0b495fffd4923750cfec187edd6f4e03dd5.
